.circleci/config.yml

```diff
@@ -528,32 +521,24 @@ jobs:
 
             # For some reason everything works fine if we use the host clang,
             # not the Xcode-bundled clang.
-            echo '[target.x86_64-apple-darwin]' >> ~/.cargo/config
-            echo 'linker = "/usr/bin/clang"' >> ~/.cargo/config
+            echo '[target.aarch64-apple-darwin]' >> ~/.cargo/config
```

Comment:
Do you see any reason that we might want to continue to test for x86 Apple hardware or is updating to Apple silicon here cover us good enough?  (not blocking, just curious if you see any need to maintain coverage over old hardware, though I guess most of us that build Glean have upgraded by now)

Reply:
For Apple aarch is where the future is.
CircleCI is dropping the x64 intel machines in the summer next year. I wanted to try and see if they work for us today. They _are_ more expensive ([twice to be precise](https://circleci.com/product/features/resource-classes/#macos)), so I might drop back to x64 for the time being.

Reply:
oh one more reason: we don't have an intel mac anymore for dev work! Which means it's extremely hard for us to debug when CI breaks on it as we won't be able to reproduce it locally correctly.